### PR TITLE
feat: add C implementation for `math/base/special/fast/uint32-sqrt`

### LINKED ISSUE #1932: [RFC]: add C implementation for `@stdlib/math/base/special/fast/uint32-sqrt`
### Description

This RFC proposes add C implementation for `@stdlib/math/base/special/fast/uint32-sqrt`

      uint32_t stdlib_base_fast_uint32_sqrt( uint32_t  x );

### Related Issues

Related issues #1917 .

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Userf1995**
Resolves #1932 

## Description

> What is the purpose of this pull request?

This pull request:

-   add C implementation for `math/base/special/fast/uint32-sqrt`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #1932 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [X] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/special/fast/uint32-sqrt/manifest.json [hunk 1]
```diff
@@ -0,0 +1,73 @@
+{
+	"options": {
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/fast/uint32-sqrt/manifest.json]: Please update the indentation to two spaces in the `manifest.json` file.

File: lib/node_modules/@stdlib/math/base/special/fast/uint32-sqrt/src/floor.c [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+#include "stdlib/math/base/special/fast/uint32_sqrt.h"
+#include <stdint.h>
+
+static const uint32_t BIT = 1073741824u;
+
+/**
+* Returns an integer square root.
+*
+* @param {uinteger32} x - input value
+* @returns {uinteger32} integer square root
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/fast/uint32-sqrt/src/floor.c]:
```suggestion
* @param x    input value
* @return     integer square root
```

File: lib/node_modules/@stdlib/math/base/special/fast/uint32-sqrt/src/floor.c [hunk 3]
```diff
@@ -0,0 +1,59 @@
+/**
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/fast/uint32-sqrt/src/floor.c]: Shouldn't this file be named e.g. `sqrt.c`? 

Notice that we would need to rename the `src` fields in the `manifest.json` file as well.

**Review by Planeshifter — CHANGES_REQUESTED** [review]
Thanks! Left a few comments and questions that need to be addressed, but otherwise this PR should be ready to land.

**Comment by Userf1995** [comment]
> Thanks! Left a few comments and questions that need to be addressed, but otherwise this PR should be ready to land.

@Planeshifter I have addressed all the issues you mentioned in your review.

**Review by Planeshifter — APPROVED** [review]
LGTM; thanks! Let's get it in.

**Pull request merged**